src/algebra/big_operators/basic.lean

```diff
@@ -947,9 +949,24 @@ lemma prod_inter_mul_prod_diff [decidable_eq α] (s t : finset α) (f : α → 
 by { convert (s.prod_piecewise t f f).symm, simp [finset.piecewise] }
 
 @[to_additive]
-lemma mul_prod_diff_singleton [decidable_eq α] {s : finset α} {i : α} (h : i ∈ s)
-  (f : α → β) : f i * (∏ x in s \ {i}, f x) = ∏ x in s, f x :=
-by { convert s.prod_inter_mul_prod_diff {i} f, simp [h] }
+lemma prod_eq_mul_prod_diff_singleton [decidable_eq α] {s : finset α} {i : α} (h : i ∈ s)
+  (f : α → β) : ∏ x in s, f x = f i * ∏ x in s \ {i}, f x :=
+by { convert (s.prod_inter_mul_prod_diff {i} f).symm, simp [h] }
```

Comment:
Could you prove this using `mul_prod_diff_singleton` so that it's clear(er) for the reader that these lemmas are closely related. Maybe also put them next to each other?

Reply:
Was this comment intended for a different line? `mul_prod_diff_singleton` no longer exists, this lemma replaces it.

Reply:
I mean https://leanprover-community.github.io/mathlib_docs/algebra/big_operators/basic.html#finset.mul_prod_diff_singleton which I believe is the `eq.symm` of this lemma?

Reply:
Oh, nevermind, I didn't look at the left column that this lemma is being removed.